src/legacy/server/kbn_server.js

```diff
@@ -199,8 +199,7 @@ export default class KbnServer {
   }
 
   applyLoggingConfiguration(settings) {
-    const config = new Config(this.config.getSchema(), settings);
-
+    const config = Config.withDefaultSchema(settings);
```

Comment:
I'm not sure what the logic inside `getSchema` does, but it doesn't return the same schema as was used to construct to config https://github.com/elastic/kibana/blob/8e9a8a84dccfa7965ce8a22362885e6cdef8b51f/src/legacy/server/config/config.js#L194-L217

Reply:
Yea, the initial legacy config is created using

https://github.com/elastic/kibana/blob/8e9a8a84dccfa7965ce8a22362885e6cdef8b51f/src/legacy/plugin_discovery/find_plugin_specs.js#L46-L48

However, just to be sure to understand, the legacy logging reloading is done from

https://github.com/elastic/kibana/blob/3ac3e7a8cce1912e19c38f2077c3daa3b46b2270/src/core/server/legacy/legacy_service.ts#L106-L110

Is that correct? The `if (this.kbnServer !== undefined)` would explain why the error is not happening during plugin discovery.

Reply:
yes, that's my understanding as well